### PR TITLE
[1] Search Bar for Store Lookup

### PR DESCRIPTION
**Pull request by tommypoa**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
- Created a search bar for store lookup
   - Clickable search results to autofill the search bar

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Related PRs
- Dependent on #15 

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
- Alignment to Figma
- Catching bugs by going through entire log-in flow
- Code quality

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
- Highlight only search string in the search output (as seen in Figma)
- Prevent user from scrolling up (thereby revealing the background)

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
- Edge case: when user types in a store that matches no stores, they have to click out of/blur the search bar (to force a "reset") and type again to search for stores.
- After searching, user has to click out of/blur the search bar before being able to click on the searched outputs

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
- Initial state
image
- Search bar on focus
image
- Search string
image
- Picked
image

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/StoreLookupScreen.js [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+  CheckInContainer,
+  CheckInContentContainer,
+  TextField,
+  SearchElement,
+  SearchBarContainer
+} from '../styled/checkin';
+
+const DismissKeyboard = ({ children }) => (
+  <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>{children}</TouchableWithoutFeedback>
+);
```
Comment (wangannie) [inline screens/StoreLookupScreen.js]: Unrelated: we should implement this in the customer store search!

File: screens/StoreLookupScreen.js [hunk 2]
```diff
@@ -50,41 +73,62 @@ export default class StoreLookupScreen extends React.Component {
...
-          {this.state.stores.map(store => {
-            return <Picker.Item label={store.storeName} value={store} key={store} />;
-          })}
-        </Picker>
-        {this.state.errorMsg ? <Text>{this.state.errorMsg}</Text> : null} */}
-        {/* <Button title="Testing Bypass" onPress={() => this._devBypass()} /> */}
-      </CheckInContainer>
+      <DismissKeyboard>
+        <CheckInContainer>
+          <CheckInContentContainer>
```
Comment (wangannie) [inline screens/StoreLookupScreen.js]: Slightly annoying to click once to dismiss the keyboard then click again to select the store I want
